_episodes_rmd/06-tidy-data.Rmd

```diff
@@ -0,0 +1,147 @@
+                         key = "day",
+                         value = "inflammatory_response",
+                         V1:V40)
+tail(dat_long)
+```
+
+Notice the row numbers? Because we gathered observations into one column, and thus one
+observation per row, the `r nrow(dat_long)` rows now are the product of the
+initial `r nrow(dat)` rows and `r ncol(dat)` columns.
+
+Because the days shouldn't remain labelled with `V`s, we also convert them to
+numbers. This works with either `as.factor()` and `as.numeric()` after another,
+or nested: 
+
+```{r inflammation-days}
+dat_long$day <- as.numeric(as.factor(dat_long$day))
+str(dat_long)
+```
+
+The structure of our inflammation dataset is now a lot tidier than before and we
+can finally package it up in an R-supported format. Rename the dataframe to some-
+thing more self-descriptive than `dat...`, using for example
+`inflammation <- dat_long` and then run `use_data(inflammation)`.
+
+> ## In case you need to "widen" a long/tidy dataset again:
+> Use the opposite of `gather()`: [`spread()`](https://tidyr.tidyverse.org/reference/spread.html).
+> You can learn more about both in the [Data Carpentry's "Data Manipulation" lesson](http://www.datacarpentry.org/R-ecology-lesson/03-dplyr.html#spreading)
+> or the ["R for Data Science" book](http://r4ds.had.co.nz/tidy-data.html#spreading).
+{: .callout}
+
+## Documenting datasets
+
+Just as a good function documentation makes your code more accessible and reusable
+for others and your future self, datasets benefit from a documentation as well.
+Please read through [r-pkgs.had.co.nz/data.html#documenting-data](http://r-pkgs.had.co.nz/data.html#documenting-data)
+and keep in mind, that the descriptive tags for datasets are not entirely the
+same as the those for function documentation.
+
+> ## Challange: Document the dataset!
+>
+> 1. Which file should we create to document our dataset?
+> 1. Which items of our dataset should be documented?
+> 1. How could the documentation file look in the end?
+>
+> > ## Solutions
+> > 
+> > 1. `R/inflammation.R`
+> > 1. `patient_ID`, `day` & `inflammatory_response`
+> > 1. Like this:
+> > ~~~
+> > #' Inflammation In Patients During Study...
+> > #'
+> > #' @source Pre-registration: \url{http://wwww.alltrials.net/study...}.
+> > #'   Method: \url{...}.
+> > #' @format A data frame with the variables/columns:
+> > #' \describe{
+> > #'  \item{patient_ID}{A factor prepresenting the different patients}
+> > #'  \item{day}{Number of days after start of the study}
+> > #'  \item{inflammatory_response}{Measured daily as described in the methods section of ...}
+> > #' }
+> > "inflammation"
+> > ~~~
+> {: .solution}
+{: .challenge}
+
+After saving the raw data, the data creation script, the tidy dataset,and
+the documentation file, what is left to do in our little personal package?
+
+> ## Challange: Last packaging steps (for now; an incomplete list)
+>
+> > ## Solutions
+> > 
+> > 1. Run `roxygen2::roxygenise()`.
```

Comment:
Or `devtools::document()`

Reply:
Because it's not mentioned anywere else in the lesson anymore, I'd refer learners to `??roxygenise` in case of problems. There, they'll find a note about when to use `devtools::document()`.